ArgumentResolver/DocumentValueResolver.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MongoDBBundle\ArgumentResolver;
+
+use Symfony\Bridge\Doctrine\ArgumentResolver\EntityValueResolver;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+/** @internal */
```

Comment:
Why It's marked as `@internal` if the original EntityValueResolver  is not `@internal`?

Reply:
Hi @pepeh, in this case we provide the feature of fetching objects from the controller automatically, I decided to mark this class as `@internal` since to me it was an implementation detail, if we could use the `EntityValueResolver`, we could remove this class.

Are you using this class directly?

Reply:
@franmomu there is option to specify value resolver manually 
`#[ValueResolver(DocumentValueResolver::class)]`
https://symfony.com/doc/current/controller/value_resolver.html#managing-value-resolvers
But if class is `@internal` IDE will show warning